PyDSS/dataset_buffer.py

```diff
@@ -0,0 +1,120 @@
+"""Contains DatasetBuffer"""
+
+import logging
+
+import h5py
+import numpy as np
+import pandas as pd
+
+
+KiB = 1024
+MiB = KiB * KiB
+GiB = MiB * MiB
+
+
+logger = logging.getLogger(__name__)
+
+
+class DatasetBuffer:
+    """Provides a write buffer to an HDF dataset to increase performance.
+    Users must call flush_data before the object goes out of scope to ensure
+    that all data is flushed.
+
+    """
+    # TODO add support for context manager, though PyDSS wouldn't be able to
+    # take advantage in its current implementation.
+
+    # The optimal number of chunks to store in memory will vary widely.
+    # The h5py docs recommend keeping chunk byte sizes between 10 KiB - 1 MiB.
+    # This attempts to support a higher-end PyDSS case. Might need to provide
+    # customization capabilities.
+    # Each element property will have one "chunk" of data in memory.
+    # Storing 50,000 element properties with a 32 KiB buffer in each of 36
+    # parallel processes would require 54 GiB of RAM.
+    MAX_CHUNK_BYTES = 32 * KiB
```

Comment:
@AadilLatif Please comment if you think another value would be better.  Or if we need to allow user customization.

Reply:
I would reccomend we keep it customizable. It would be easy to change if we run into corner casses. I suspect the chosen value shuld be valid for 95% of simulations we do.